src/did/resolver.ts

```diff
@@ -1,29 +1,38 @@
-import { Resolver, DIDDocument } from "did-resolver";
+import { Resolver, DIDDocument, PublicKey } from "did-resolver";
 import { getResolver as ethrGetResolver } from "ethr-did-resolver";
 import { getResolver as webGetResolver } from "web-did-resolver";
 import NodeCache from "node-cache";
 import { INFURA_API_KEY } from "../config";
 
+export interface EthrResolverConfig {
+  networks: Array<{
+    name: string;
+    registry?: string;
+    rpcUrl: string;
+  }>;
+}
+
 const providerConfig = {
   networks: [{ name: "mainnet", rpcUrl: `https://mainnet.infura.io/v3/${INFURA_API_KEY}` }],
 };
 
 const didResolutionCache = new NodeCache({ stdTTL: 5 * 60 }); // 5 min
 
-export const resolver = new Resolver({
-  ...ethrGetResolver(providerConfig),
-  ...webGetResolver(),
-});
+const defaultResolver = new Resolver({ ...ethrGetResolver(providerConfig), ...webGetResolver() });
+
+export const createResolver = (ethrResolverConfig: EthrResolverConfig): Resolver => {
```

Comment:
the naming bother me a bit, because it's a generic name to create a custom ethr did. Suggestion
- rename to createEthrDidResolver
- make the first param an object, with `ethrResolverConfig` as optional and then add or not `ethrGetResolver(ethrResolverConfig)` depending on the presence of the option.

Reply:
1) The name createEthrDidResolver could be confusing, as it creates a resolver both for ethr method and web method, like the default resolver.

2) If the custom ethr config is optional, I guess that if it's not passed as args, I should return the default resolver. So, what do you think of:

```
export const createResolver = (ethrResolverConfig?: EthrResolverConfig): Resolver => {
  return ethrResolverConfig
    ? new Resolver({ ...ethrGetResolver(ethrResolverConfig), ...webGetResolver() })
    : defaultResolver;
};
```

Reply:
Yes that will do, but also I think it would be better to use an object, so that it will be easier to add more resolver configuration later

```ts
export const createResolver = ({ethrResolverConfig}: {ethrResolverConfig?: EthrResolverConfig}): Resolver => {
  return ethrResolverConfig
    ? new Resolver({ ...ethrGetResolver(ethrResolverConfig), ...webGetResolver() })
    : defaultResolver;
};
```

Reply:
done